extensions/stats/plugin.cc

```diff
@@ -188,8 +188,7 @@ const wasm::common::NodeInfo& NodeInfoCache::getPeerById(
   return cache_[peer_id];
 }
```

Comment:
Turns out Stats plugin does not have a context factory... It surprises me that the module could be loaded successfully, shouldn't it fail loading?

I tried to add context factory, but it does not compile because of some undefined symbols.

Reply:
Can you send me your changes (repo branch) ?

Reply:
If you have the list of undefined symbols handy could you send it to me ?

Reply:
This is the diff:
```
diff --git a/extensions/stats/plugin.h b/extensions/stats/plugin.h
index 338dc6d..4f1d54b 100644
--- a/extensions/stats/plugin.h
+++ b/extensions/stats/plugin.h
@@ -414,11 +414,11 @@ class PluginContext : public Context {
 
 #ifdef NULL_PLUGIN
 NULL_PLUGIN_ROOT_REGISTRY;
+#endif
 
 static RegisterContextFactory register_Stats(
     CONTEXT_FACTORY(Stats::PluginContext),
     ROOT_FACTORY(Stats::PluginRootContext));
-#endif
 
 }  // namespace Stats
```

And this is the error:
```
error: undefined symbol: _ZN4Wasm6Common19extractNodeMetadataERKN6google8protobuf6StructEPN4wasm6common8NodeInfoE
warning: To disable errors for undefined symbols use `-s ERROR_ON_UNDEFINED_SYMBOLS=0`
error: undefined symbol: _ZN4Wasm6Common23populateHTTPRequestInfoEbPNS0_11RequestInfoE
error: undefined symbol: _ZN4Wasm6Common24extractLocalNodeMetadataEPN4wasm6common8NodeInfoE
error: undefined symbol: _ZN4absl13hash_internal10CityHash32EPKcm
error: undefined symbol: _ZN4absl18container_internal10SampleSlowEPx
error: undefined symbol: _ZN4absl18container_internal12UnsampleSlowEPNS0_14HashtablezInfoE
error: undefined symbol: _ZN4absl18container_internal16RecordInsertSlowEPNS0_14HashtablezInfoEmm
error: undefined symbol: _ZN4absl18container_internal21ShouldInsertBackwardsEmPa
error: undefined symbol: _ZN4absl5Mutex4LockEv
error: undefined symbol: _ZN4absl5Mutex6UnlockEv
error: undefined symbol: _ZN4wasm6common8NodeInfo12InternalSwapEPS1_
error: undefined symbol: _ZN4wasm6common8NodeInfoC1Ev
error: undefined symbol: _ZN4wasm6common8NodeInfoC2Ev
error: undefined symbol: _ZN4wasm6common8NodeInfoD1Ev
error: undefined symbol: _ZN5stats12PluginConfigC1Ev
error: undefined symbol: _ZN5stats12PluginConfigD1Ev
error: undefined symbol: _ZN4absl13hash_internal13CityHashState5kSeedE
Error: Aborting compilation due to previous errors
shared:ERROR: '/root/emsdk/node/12.9.1_64bit/bin/node /root/emsdk/fastcomp/emscripten/src/compiler.js /tmp/tmpU6rAdZ.txt /sdk/proxy_wasm_intrinsics.js /root/emsdk/fastcomp/emscripten/src/library_pthread_stub.js' failed (1)
Makefile:10: recipe for target 'plugin.wasm' failed
```
Some of the symbols are from pb.cc files which should be compiled together, some of them needs more absl targets to be compiled with.

Reply:
Builds now.